### PR TITLE
only require rustfmt and update_lints for version bumps

### LINKED ISSUE #1487: Remove `rustfmt` from travis builds
I vote to remove `rustfmt` from travis.

I find it really annoying, blindly applying `rustfmt` suggestions does not make the code clearer in lots of instances and even often make it worst ([recent example](https://github.com/Manishearth/rust-clippy/pull/1486#issuecomment-275847501)).
It also seems really hard to make `rustfmt` happy both locally and on travis ([1](https://github.com/Manishearth/rust-clippy/pull/1465#issuecomment-274332709), [2](https://github.com/Manishearth/rust-clippy/pull/1486#issuecomment-275847633), [3](https://github.com/Manishearth/rust-clippy/pull/1457#issuecomment-274138969)).

### PR DESCRIPTION
**Pull request by oli-obk**
fixes #1487

### CODE REVIEW

File: PUBLISH.md [hunk 1]
```diff
@@ -1,15 +1,14 @@
 Steps to publish a new clippy version
 
-1. `cargo test`.
 - Bump `package.version` in `./Cargo.toml` (no need to manually bump `dependencies.clippy_lints.version`).
-- Run `./util/update_lints.py`.
+- Run `./pre_publish.sh`
 - Write a changelog entry.
-- Commit `./Cargo.toml`, `./clippy_lints/Cargo.toml` and `./CHANGELOG.md`.
+- Review and commit all changed files
```
Comment (mcarton) [inline PUBLISH.md]: I prefer to have one commit just for the version bump itself,that is the `Cargo.toml`s and the changelog.

File: pre_publish.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -e
+
+cd clippy_lints && cargo fmt && cd ..
+cargo fmt
```
Comment (mcarton) [inline pre_publish.sh]: Can it add `--write-mode=overwrite` if the repo is clean? I hate those .bk files

File: PUBLISH.md (hunk 1 above)
Comment (oli-obk) [inline PUBLISH.md]: I changed the script to do this for us

File: pre_publish.sh (hunk 2 above)
Comment (oli-obk) [inline pre_publish.sh]: done

File: pre_publish.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+git status --short | sort | grep -v README.md | grep -v helper.txt > helper.txt
+
+# abort if the files differ
+diff "publish.files" "helper.txt"
+
+rm helper.txt
+
+# add all changed files
+git add .
+git commit -m "version bump"
```
Comment (mcarton) [inline pre_publish.sh]: This definitely needs to contain the version then. [Also, this fails rules 3 and 5.](https://chris.beams.io/posts/git-commit/)

Comment (oli-obk) [inline pre_publish.sh]: we tag it later, do we really need the version in there?

**Comment by oli-obk** [comment]
any further complaints? or can we merge this?

**Comment by oli-obk** [comment]
should we move dogfood tests to version bumps, too?

**Comment by mcarton** [comment]
No. They are good to see the effect of a new lint on a real codebase. See #1556 for example, dogfood shows quite a few nice examples of what the lint does.

**Comment by llogiq** [comment]
I'm also bullish in dogfooding.

**Comment by llogiq** [comment]
Yay! \o/

**Pull request merged**